website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+---
+title: LLM as judge
+description: Perform LLM Evaluations with custom metrics
+slug: llm-as-judge
+authors: [pedro-azevedo,rahul-pandey]
+tags: [genai,mlflow-evalaute]
+thumbnail: img/blog/llm-as-judge.png
+---
+
+In this blog post, we'll dive on a journey to revolutionize how we evaluate language models. We'll explore the power of MLflow Evaluate and harness the capabilities of Large Language Models (LLMs) as impartial judges. By the end, you'll learn how to create custom metrics, implement LLM-based evaluation, and apply these techniques to real-world scenarios. Get ready to transform your model assessment process and gain deeper insights into your AI's performance!
+
+## The Challenge of Evaluating Language Models
+
+Imagine you're part of a global travel agency, "WorldWide Wandercorp," that's expanding its reach to Spanish-speaking countries. Your team has developed an AI-powered translation system to help create culturally appropriate marketing materials and customer communications. However, as you begin to use this system, you realize that traditional evaluation metrics fall short in capturing the nuances of language translation, especially when it comes to preserving cultural context and idiomatic expressions. You want to understand the materials in which your translation is failing so you can improve the model or manually translate them.
+
+This is where MLflow Evaluate and LLMs as judges come into play. Let's dive into how these tools can help solve Worldwide WanderAgency's challenges and improve their AI-driven communication strategy.
+
+## Introducing MLflow Evaluate
+
+The traveling agency needs a way to evaluate the outputs of it's LLM translation model.
+MLflow Evaluate is a powerful function within the MLflow ecosystem that allows for comprehensive model assessment. It supports both built-in metrics and custom metrics, making it an ideal tool for evaluating complex language tasks. With MLflow Evaluate, you can:
+
+- Evaluate models against multiple metrics simultaneously
+- Use pre-defined metrics for specific model types (e.g., question-answering, text-summarization and pure text)
+- Create custom metrics, including those that use LLMs as judges
```

Comment:
Let's mention both of the APIs for creating custom GenAI metrics: 

[mlflow.metrics.genai.make_genai_metric()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html#mlflow.metrics.genai.make_genai_metric)
 and
[mlflow.metrics.genai.make_genai_metric_from_prompt()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html#mlflow.metrics.genai.make_genai_metric_from_prompt)